lib/URI.pm

```diff
@@ -100,8 +114,16 @@ sub _init
 sub _fix_uric_escape_for_host_part {
   return if HAS_RESERVED_SQUARE_BRACKETS;
   return if $_[0] !~ /%/;
+  return if $_[0] =~ m,^(?:$URI::schemes_without_host_part_re):,os;
+
+  # until a scheme specific handler is available, fall back to previous behavior of v5.10 (i.e. 'mailto:')
+  if ($_[0] =~ m,^(?:$URI::fallback_schemes_re):,os) {
```

Comment:
Could we have a different regex delimiter here? I find the comma hard to read. The curly braces are pretty standard.

So:

```perl
 m{^(?:$URI::fallback_schemes_re):}os
```

Reply:
Sure. I like them better too. I wanted to stick to the given style and not add another one.
Also, the `o` flag doesn't seem to be useful nowadays. I rather use `qr//` in modern code.
Both might break legacy code (or does not work with ancient versions of Perl), so I thought - better don't touch it ;-)
Will use curlies.